modules/sor/sor.gql

```diff
@@ -173,6 +195,55 @@ enum GqlSorSwapType {
     EXACT_OUT
 }
 
+type GqlSorCallData {
+    """
+    The call data that needs to be sent to the RPC
+    """
+    callData: String!
+    """
+    The target contract to send the call data to
+    """
+    to: String!
+    """
+    Value in ETH that needs to be sent for native swaps
+    """
+    value: BigDecimal!
+    """
+    Minimum amount received for exact in orders
+    """
+    minAmountOutRaw: String
+    """
+    Maximum amount to be sent for exact out orders
+    """
+    maxAmountInRaw: String
+}
+
+"""
+Inputs for the call data to create the swap transaction. If this input is given, call data is added to the response.
+"""
+input GqlSwapCallDataInput {
+    """
+    The max slippage in percent.
+    """
+    slippagePercentage: String!
+    """
+    How long the swap should be valid, provide a timestamp. "999999999999999999" for infinite. Default: infinite
+    """
+    deadline: Int
+    """
+    Who sends the input amount.
+    """
+    sender: String!
+    """
+    Who receives the output amount.
+    """
+    receiver: String!
+    """
+    If a native token is either input or output token.
+    """
+    wethIsEth: Boolean!
```

Comment:
This naming doesn't sound ideal.. i guess it's coming from the sdk? from the comment i would assume that i have to set it when tokenIn / out is native, but cannot it be easily derived? Is it rather about expected wrapping / unwrapping? I'm thinking if we can make it seamless somehow.

Reply:
you're right